third_party/terraform/tests/resource_google_project_test.go

```diff
@@ -200,8 +200,10 @@ func testAccCheckGoogleProjectExists(r, pid string) resource.TestCheckFunc {
 			return fmt.Errorf("No ID is set")
 		}
 
-		if rs.Primary.ID != pid {
-			return fmt.Errorf("Expected project %q to match ID %q in state", pid, rs.Primary.ID)
+		parts := strings.Split(rs.Primary.ID, "/")
+		foundId := parts[len(parts)-1]
+		if foundId != pid {
```

Comment:
I'd slightly prefer if we changed the expected value instead of the state value, but not a blocker